### PR TITLE
Add git init and precommit hook for eslint

### PR DESCRIPTION
**Pull request by ceejtron**
As per #43 this adds precommit hooks for eslint. In addition, this adds a new question for initializing a git repository since precommit hooks can't be automatically set up unless a git repository has already been initialized.

### CODE REVIEW

**Comment by italopessoa** [comment]
Well done @ceejtron. Seems that is everything ok. Let's wait for the reviewers. Thank you for your contribution ❤️

File: hooks/eslint/index.js [hunk 1]
```diff
@@ -22,7 +22,18 @@ const func = (cwd, folderName) => {
 }
 
 const addEslintFileSuccess = (cwd, folderName) => {
-  const command = `npm i eslint eslint-config-airbnb babel-eslint --save-dev`
+  const eslintDependencies = [
```
Comment (jouderianjr) [inline hooks/eslint/index.js:25]: Nice Refactor mate

File: hooks/git/index.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  console.log('\n\n')
+
+  const command = 'git init'
+  const terminalOpts = {
+    cwd: `${cwd}/${folderName}`,
+    shell: true,
+    stdio:'inherit',
+  }
+
+  return spawn(command, [], terminalOpts)
```
Comment (jouderianjr) [inline hooks/git/index.js]: So, We tried to find if has some issue on create-react-app related with that and I've found this PR https://github.com/facebookincubator/create-react-app/pull/1288

So, looks like that will be added at some moment, what do you think about test if already has a git started? maybe just test if exists .git folder.

**Review by jouderianjr — CHANGES_REQUESTED** [review]
Hey, nice work! I'm pretty happy to see a new hook 💃

**Comment by paulonotz0r** [comment]
Nice, @ceejtron! This is really cool!

In addition, can you add `commit lint` too? You can use the same as used on ifactory-solutions/inside-client#8.

Thank you for the contribution.

**Comment by ceejtron** [comment]
Added a check for .git as well as a commitlint hook

**Review by paulonotz0r — APPROVED** [review]
Great!

File: hooks/git/index.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    stdio:'inherit',
+  }
+
+  if (gitExists) {
+    console.log('\n\n')
+    console.log(chalk.cyan('Git repository already initialized. Skipping...'))
+    console.log('\n\n')
+    return Promise.resolve(true)
+  }
+  else {
```
Comment (jouderianjr) [inline hooks/git/index.js]: Could you please use else in the same line? 😄

File: hooks/commitlint/index.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+      const command = `npm i ${commitlintDependencies.join(' ')} --save-dev`
+      const terminalOpts = {
+        cwd: `${cwd}/${folderName}`,
+        shell: true,
+        stdio:'inherit',
+      }
+
+      return spawn(command, [], terminalOpts)
+    })
+    .then(() => {
```
Comment (jouderianjr) [inline hooks/commitlint/index.js]: You can do that as one line 
` .then(() => loadPackageJsonFromPath(`${cwd}/${folderName}/package.json`))`

**Review by jouderianjr — CHANGES_REQUESTED** [review]
Just 2 minor things, then you will merge it! 💃

**Comment by ceejtron** [comment]
@jouderianjr I pushed the changes you asked for

**Review by jouderianjr — APPROVED** [review]

**Comment by jouderianjr** [comment]
Awesome @ceejtron thanks for your work 💃

**Pull request merged**